### ISSUE TITLE
Enable specifying list type in computed fields

### ISSUE THREAD

**Issue by lucperkins** [contentlayerdev/contentlayer#277]
As far as I can tell, it isn't possible to specify the `of` type of lists in computed fields the way that you can in non-computed fields. Thus, this is possible:

```typescript
fields: {
  tags: {
    type: "list",
    of: { type: "string" },
    default: []
  }  
}
```

But this is not:

```typescript
computedFields: {
  segments: {
    type: "list",
    of: { type: "string" },
    resolve: (doc) => { ... }
  }
}
```

There may be reasons to explicitly disallow this, and if so I'd be curious why! Otherwise, this feels like an oversight to me.

**Comment by schickling** [comment]
I agree there are definitely some gaps in regards to which types can be used for computed fields. This will be addressed before 1.0 is released. (Also see #149)

**Comment by lucperkins** [comment]
@schickling Sounds great. I look forward to it!
